msteams-platform/toolkit/configure-message-extension-capability.md

```diff
@@ -0,0 +1,773 @@
+
+   ```json
+   {
+   "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+   "contentVersion": "1.0.0.0",
+   "parameters": {
+     "resourceBaseName": {
+       "value": "tab${{RESOURCE_SUFFIX}}"
+     },
+     "webAppSku": {
+       "value": "B1"
+     },
+     "botAadAppClientId": {
+       "value": "${{BOT_ID}}"
+     },
+     "botAadAppClientSecret": {
+       "value": "${{SECRET_BOT_PASSWORD}}"
+     },
+     "botDisplayName": {
+       "value": "bot"
+     },
+     "resourceBaseName2":{
+       "value": "bot${{RESOURCE_SUFFIX}}"
+     }
+   }
+   ```
+
+1. Add `botAadApp/create` action under the provision section in the `teamsapp.yml` file.
+1. Add the following code in the **deploy** section:
+
+   ```yml
+   deploy:
+     - uses: cli/runNpmCommand # Run npm command
+       with:
+         args: install
+     - uses: cli/runNpmCommand # Run npm command
+       with:
+         args: run build
+     # Deploy bits to Azure Storage Static Website
+     - uses: azureAppService/zipDeploy
+       with:
+         workingDirectory: ./tab
+         # Deploy base folder
+         artifactFolder: .
+         # Ignore file location, leave blank will ignore nothing
+         ignoreFile: .webappignore
+         # The resource id of the cloud resource to be deployed to.
+         # This key will be generated by arm/deploy action automatically.
+         # You can replace it with your existing Azure Resource id
+         # or add it to your environment variable file.
+         resourceId: ${{TAB_AZURE_APP_SERVICE_RESOURCE_ID}}
+     - uses: azureAppService/zipDeploy
+       with:
+         workingDirectory: ./bot
+         # Deploy base folder
+         artifactFolder: .
+         # Ignore file location, leave blank will ignore nothing
+         ignoreFile: .webappignore
+         # The resource id of the cloud resource to be deployed to.
+         # This key will be generated by arm/deploy action automatically.
+         # You can replace it with your existing Azure Resource id
+         # or add it to your environment variable file.
+         resourceId: ${{BOT_AZURE_APP_SERVICE_RESOURCE_ID}}
+   ```
+
+   For more information, see the [sample project](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/hello-world-bot-with-tab).
+
+1. Select **Command Palette...** under the **View** option or **Ctrl+Shift+P**.
+1. Enter `Teams: Provision` to apply the bicep to Azure.
+1. Enter `Teams: Deploy` to deploy your tab app code to Azure.
+1. Select **Run and Debug Activity Panel** and select **Launch Remote (Edge)** or **Launch Remote (Chrome)**.
+1. Press **F5** to debug and preview your Teams app.
+
+# [Add message extension to bot app](#tab/botapp)
+
+You can add message extension to a bot app more easily than adding it to a tab app as message extension use the bot support framework.
+
+To add a message extension bot app to Teams follow these steps:
```

Comment:
```suggestion
To add a message extension to bot app follow these steps:
```

Reply:
Updated as: To add a message extension to a bot app, follow these steps: